astronomer/providers/dbt/cloud/triggers/dbt.py

```diff
@@ -0,0 +1,104 @@
+import asyncio
+import time
+from typing import Any, AsyncIterator, Dict, Optional, Tuple
+
+from airflow.providers.dbt.cloud.hooks.dbt import DbtCloudJobRunStatus
+from airflow.triggers.base import BaseTrigger, TriggerEvent
+
+from astronomer.providers.dbt.cloud.hooks.dbt import DbtCloudHookAsync
+
+
+class DbtCloudRunJobTrigger(BaseTrigger):
+    """
+    DbtCloudRunJobTrigger is triggered with run id and account id, makes async Http call to dbt and get the status
+    for the submitted job with run id in polling interval of time.
+
+    :param conn_id: The connection identifier for connecting to Dbt.
+    :param run_id: The ID of a dbt Cloud job.
+    :param end_time: Time in seconds to wait for a job run to reach a terminal status. Defaults to 7 days.
+    :param account_id: The ID of a dbt Cloud account.
+    :param poll_interval:  polling period in seconds to check for the status.
+    """
+
+    def __init__(
+        self,
+        conn_id: str,
+        run_id: int,
+        end_time: float,
+        poll_interval: float,
+        account_id: Optional[int],
```

Comment:
```suggestion
        account_id: int,
```

Reply:
This `account_id` can be optional as it is in sync operator if it is not passed from operator it takes it from connection object